var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -378,6 +392,44 @@ def install(self, spec, prefix):
                 '--with-scalapack=0'
             ])
 
+        # For the moment, HPDDM does not work as a dependency
+        # using download instead
+        if '+hpddm' in spec:
+            options.extend(['--download-hpddm'])
```

Comment:
Ah - circular dependency ...

Reply:
Actually, it seems to be a problem related to the way the dependency is imported into PETSc, I'll talk in more details with the developer.

Reply:
Yeah - its unavoidable..